docs/console/tutorials/project-infrastructure/project-configuration.md

```diff
@@ -0,0 +1,396 @@
+      #   echo "INFO: GCP authentication configured."
+      # else
+      #   echo "ERROR: GCP_SA_KEY_JSON CI/CD variable not set. Cloud operations will likely fail."
+      #   exit 1
+      # fi
+      # --- END OF YOUR AUTHENTICATION SCRIPT ---
+
+variables:
+  TF_IN_AUTOMATION: "true"
+  TF_WORKSPACE: "default"
+  GIT_DEPTH: 10
+
+.terraform_cache: &terraform_cache_definition
+  cache:
+    key: "${CI_PROJECT_ID}-${CI_COMMIT_REF_SLUG}-${TF_WORKSPACE}-terraform"
+    paths:
+      - .terraform
+      - .terraform.lock.hcl
+    policy: pull-push
+
+stages:
+  - validate
+  - plan
+  - apply
+
+validate:
+  stage: validate
+  extends: .terraform_cache
+  script:
+    - terraform init -reconfigure -input=false -no-color
+    # Puoi inserire qui script per analisi statica o linter aggiuntivi prima di `fmt` e `validate`
+    # Esempio: tfsec . --no-color
+    - terraform fmt -check -recursive -diff -no-color
+    - terraform validate -no-color
+  rules:
+    - if: '$CI_PIPELINE_SOURCE == "merge_request_event"'
+    - if: '$CI_COMMIT_BRANCH == $CI_DEFAULT_BRANCH'
+
+plan:
+  stage: plan
+  extends: .terraform_cache
+  script:
+    - terraform init -reconfigure -input=false -no-color
+    - |
+      # Assicura che il workspace di destinazione esista
+      if ! terraform workspace select "${TF_WORKSPACE}" -no-color; then
+        terraform workspace new "${TF_WORKSPACE}" -no-color
+      fi
+    # Qui potresti aggiungere script per generare file .tfvars dinamici o altre preparazioni
+    # Esempio: ./scripts/generate-tfvars.sh > environment.auto.tfvars
+    - terraform plan -out=tfplan -input=false -no-color
+  artifacts:
+    paths:
+      - tfplan
+    expire_in: 1 day
+  rules:
+    - if: '$CI_PIPELINE_SOURCE == "merge_request_event"'
+    - if: '$CI_COMMIT_BRANCH == $CI_DEFAULT_BRANCH'
+
+apply:
+  stage: apply
+  extends: .terraform_cache
+  script:
+    - terraform init -reconfigure -input=false -no-color
+    - terraform workspace select "${TF_WORKSPACE}" -no-color
+    # Comandi personalizzati pre-apply (es. notifiche, ultimi controlli di policy) potrebbero andare qui
+    # Esempio: ./scripts/notify-pre-apply.sh
+    - terraform apply -auto-approve "tfplan" -input=false -no-color
+    # Comandi personalizzati post-apply (es. aggiornamento CMDB, test di fumo, notifiche) qui
+    # Esempio: ./scripts/run-smoke-tests.sh
+  dependencies:
+    - plan
+  rules:
+    - if: '$CI_COMMIT_BRANCH == $CI_DEFAULT_BRANCH'
+      when: manual
+      allow_failure: false
+
+```
+
+## Infrastructure Resource Repository
+
+As already mentioned, the infrastructure resource repository is a git repository that contains the terraform code of the resource and the pipelines needed to deploy it.
+
+Here is a link to the "TFHello World" repository which contains example code useful for this how-to: [TFHello World](https://git.tools.mia-platform.eu/clients/mia-platform/platform-development/infrastructure-playground/tf-hello-world)
```

Comment:
you can't share this repo, this is a private repo only mia-platform employees have access to 

moreover this is something that we have built for r&d purposes, not something that is meant to be a gold standard for sure